pkg/endpoint/policy.go

```diff
@@ -568,46 +544,46 @@ func (e *Endpoint) regeneratePolicy(owner Owner, opts models.ConfigurationMap) (
 		e.getLogger().Debug("regeneration of L3 (CIDR) policy caused policy change")
 	}
 
-	// no failures after this point
-
-	optsChanged := e.updateAndOverrideEndpointOptions(owner, opts)
+	e.computeDesiredPolicyMapState(repo)
 
-	e.computeDesiredPolicyMapState(owner, labelsMap, repo)
 	// If we are in this function, then policy has been calculated.
 	if !e.policyCalculated {
 		e.getLogger().Debug("setting PolicyCalculated to true for endpoint")
 		e.policyCalculated = true
 		// Always trigger a regenerate after the first policy
 		// calculation has been performed
-		optsChanged = true
+		forceRegeneration = true
 	}
 
 	if e.forcePolicyCompute {
-		optsChanged = true           // Options were changed by the caller.
+		forceRegeneration = true     // Options were changed by the caller.
 		e.forcePolicyCompute = false // Policies just computed
 		e.getLogger().Debug("Forced policy recalculation")
 	}
 
+	// Set the revision of this endpoint to the current revision of the policy
+	// repository.
 	e.nextPolicyRevision = revision
 
-	if !owner.DryModeEnabled() {
-		e.syncPolicyMapController()
-	}
-
 	// If no policy or options change occurred for this endpoint then the endpoint is
 	// already running the latest revision, otherwise we have to wait for
 	// the regeneration of the endpoint to complete.
 	policyChanged := l3PolicyChanged || l4PolicyChanged
 
 	e.getLogger().WithFields(logrus.Fields{
 		"policyChanged":       policyChanged,
-		"optsChanged":         optsChanged,
 		"policyRevision.next": e.nextPolicyRevision,
+		"forcedRegeneration":  forceRegeneration,
 	}).Debug("Done calculating policy")
 
-	needToRegenerateBPF := optsChanged || policyChanged || e.nextPolicyRevision > e.policyRevision
+	// If the policy changed, or the revision of the policy repository has changed
+	// we return true. It is possible that the endpoint's next policy revision
+	// is the same as the endpoint's current policy revision; this indicates
+	// that no new rules have been added in the policy repository; if policy
+	// hasn't changed, and no new rules were added, then
```

Comment:
`then return false`, perhaps?

Reply:
Sigh, I missed this comment - will update the comment once build finishes, and will run gofmt locally and paste output to confirm I don't break master. Given that it's only a comment update, I won't run tests again provided I didn't break anything in my latest upload.